src/compute-client/src/controller.rs

```diff
@@ -1041,6 +1041,31 @@ where
             instance.call(Instance::maintain);
         }
     }
+
+    /// Allow writes for the specified collections on `instance_id`.
+    ///
+    /// If this controller is in read-only mode, this is a no-op.
+    pub fn allow_writes(
+        &mut self,
+        instance_id: ComputeInstanceId,
+        collection_ids: Vec<GlobalId>,
+    ) -> Result<(), CollectionUpdateError> {
+        if self.read_only {
+            tracing::info!("Skipping allow_writes in read-only mode");
```

Comment:
This has the potential to be very noisy, since we call this method once for every collections currently. Make this a `debug` instead?

Also, a nit but: Having a batched command (`Vec<GlobalId>` argument, instead of `GlobalId`) makes the code slightly less straightforward, and in this case also slightly less efficient because we are allocating `Vec`s even though all call sites only pass a single argument.